dex/src/main/scala/com/wavesplatform/dex/settings/OrderFeeSettings.scala

```diff
@@ -83,17 +89,55 @@ object OrderFeeSettings {
   final case class CompositeSettings(
     default: OrderFeeSettings,
     custom: Map[AssetPair, OrderFeeSettings] = Map.empty,
+    customAssets: Option[CompositeSettings.CustomAssetsSettings] = None,
     discount: Option[CompositeSettings.DiscountAssetSettings] = None,
     zeroFeeAccounts: Set[PublicKey] = Set.empty
   ) extends OrderFeeSettings {
 
     def getOrderFeeSettings(assetPair: AssetPair): OrderFeeSettings =
-      custom.getOrElse(assetPair, default)
+      custom.get(assetPair).orElse {
+        customAssets.flatMap(_.getSettings(assetPair))
+      }.getOrElse(default)
+
+    def getAllPairs: Map[AssetPair, OrderFeeSettings] =
+      customAssets.fold(Map.empty[AssetPair, OrderFeeSettings])(
+        _.settingsMap
+      ) ++ custom // custom must override any values from customAssets
 
   }
 
   object CompositeSettings extends ConfigReaders {
 
+    final case class CustomAssetsSettings(assets: Set[Asset], settings: OrderFeeSettings)(validator: AssetPairValidator) {
```

Comment:
let's avoid curring in case classes. just use
```
final case class CustomAssetsSettings(assets: Set[Asset], settings: OrderFeeSettings, validator: AssetPairValidator)
```

Reply:
and CustomAssetsSettings(assets, settings, _) for lambda variant